packages/ui-components/src/components/DataLibrary/ResultsList.js

```diff
@@ -44,12 +44,16 @@ export const ResultsList = ({
 
             if (allowAddMultipleTimes) {
               return (
-                <SelectableMultipleTimesOption key={index} option={option} onSelect={onSelect} />
+                <SelectableMultipleTimesOption
+                  key={option.code}
+                  option={option}
+                  onSelect={onSelect}
+                />
               );
             } else {
               return (
                 <SelectableOption
-                  key={index}
+                  key={option.code}
```

Comment:
When you use index of an array as a key, React will optimize and not render as expected. https://stackoverflow.com/questions/46735483/error-do-not-use-array-index-in-keys